site/e2e/tests/roles.spec.ts

```diff
@@ -0,0 +1,157 @@
+import { type Page, expect, test } from "@playwright/test";
+import {
+	createOrganization,
+	createOrganizationMember,
+	setupApiCalls,
+} from "../api";
+import { license, users } from "../constants";
+import { login, requiresLicense } from "../helpers";
+import { beforeCoderTest } from "../hooks";
+
+test.beforeEach(async ({ page }) => {
+	beforeCoderTest(page);
+});
+
+type AdminSetting = (typeof adminSettings)[number];
+
+const adminSettings = [
+	"Deployment",
+	"Organizations",
+	"Healthcheck",
+	"Audit Logs",
+] as const;
+
+async function hasAccessToAdminSettings(page: Page, settings: AdminSetting[]) {
+	// Organizations and Audit Logs both require a license to be visible
+	const visibleSettings = settings.filter(
+		(it) => Boolean(license) || (it !== "Organizations" && it !== "Audit Logs"),
+	);
```

Comment:
Nit: I feel like this might be a little more clear that if we have a license, we don't need to worry about granularity as much:
```suggestion
	const visibleSettings = license
	  ? settings
	  : settings.filter((it !== "Organizations" && it !== "Audit Logs"));
```